clients/da-vinci-client/src/main/java/com/linkedin/davinci/StoreBackend.java

```diff
@@ -154,7 +154,13 @@ synchronized CompletableFuture<Void> subscribe(
       // Recreate store config that was potentially deleted by unsubscribe.
       config.store();
     }
+
+    ComplementSet<Integer> unassignedPartitionSet = ComplementSet.newSet(subscription);
```

Comment:
Upon bootstrapping, it seems that there is a setting to throw a VeniceException when a version is present. 
Bootstrapping is invoked in DaVinciBackend. 
Additionally, subscription of partitions is invoked in Da Vinci Client where set of partitions can be passed into the subscribe function directly.

Reply:
What can happen is that when you call `unsubscribe` [subscription](https://github.com/linkedin/venice/blob/ce23e3895159478b5305714644f9bbeb85c1251e/clients/da-vinci-client/src/main/java/com/linkedin/davinci/StoreBackend.java#L198-L211) can be empty, causing it to delete the current version.

Also, I don't think `subscribe` is the right place to perform the validation. If a user is subscribing in a for loop, not everything would have been added to subscription yet on the first n-1 invocations. It would be better to do this validation somewhere we can guarantee all subscriptions have been performed. `DaVinciBackend::bootstrap` might be a good place.